Support anydata for client data binding (#991)
ballerina-platform/module-ballerina-http: docs/spec/spec.md

##### 2.3.4.4. Payload parameter

The payload parameter is used to access the request payload during the resource invocation. When the payload param is 

> defined with @http:Payload annotation, the listener deserialize the inbound request payload based on the mime type 

Reviewer: ```suggestion
defined with @http:Payload annotation, the listener deserialize the inbound request payload based on the media type 
```

Reviewer: It seems media type is the correct terms which is formally known as MIME type.

which retrieved by the `Content-type` header of the request. The data binding happens thereafter considering the 

> parameter type. The type of payload param can be one of the `anytype`. If the header is not present or not a 

Reviewer: ```suggestion
parameter type. The type of payload parameter can be one of the `anytype`. If the header is not present or not a 
```

> standard header, the binding type is inferred by the param type.

Reviewer: ```suggestion
standard header, the binding type is inferred by the parameter type.
```

> Following table explains the compatible `anydata` types with each common mime type. In the absence of a standard mime 

Reviewer: ```suggestion
Following table explains the compatible `anydata` types with each common media type. In the absence of a standard mime 
```

> type, the binding type is inferred by the payload param type itself. If the type is not compatible with the mime type,

Reviewer: ```suggestion
type, the binding type is inferred by the payload param type itself. If the type is not compatible with the media type,
```

infer the expected payload type from the LHS variable type. This is called as client payload binding support where the 
inbound response payload is accessed and parse to the expected type in the method signature. It is easy to access the
payload directly rather manipulation `http:Response` using its support methods such as `getTextPayload()`, ..etc.

> Client data binding supports `anydata` where the payload is deserialized based on the mime type before binding it 

Reviewer: ```suggestion
Client data binding supports `anydata` where the payload is deserialized based on the media type before binding it 
```

to the required type. Similar to the service data binding following table explains the compatible `anydata` types with 

> each common mime type. In the absence of a standard mime type, the binding type is inferred by the payload param type 

Reviewer: ```suggestion
each common mime type. In the absence of a standard media type, the binding type is inferred by the payload param type 
```

> each common mime type. In the absence of a standard mime type, the binding type is inferred by the payload param type 

Reviewer: ```suggestion
each common mime type. In the absence of a standard media type, the binding type is inferred by the payload parameter type 
```

> itself. If the type is not compatible with the mime type, error is returned.

Reviewer: ```suggestion
itself. If the type is not compatible with the media type, error is returned.
```
